includes/discovery/sensors/rittal-cmc-iii-sensors.inc.php

```diff
@@ -36,15 +38,34 @@
     $var_type = $entry['cmcIIIVarType'];
     $sensor_id = count($cmc_iii_sensors);
 
-    if ($cmc_iii_sensors[$sensor_id]['name'] != $sensor_name) {
+    $index_r = explode('.', $index);
+    if (count($index_r) > 1) {
+        $current_index_prefix = $index_r[0];
+    }
+
+    if ($cmc_iii_sensors[$sensor_id]['name'] != $sensor_name || $last_index_prefix != $current_index_prefix) {
         if ($sensor_id == 0) {
             $sensor_id = 1;
         } else {
             $sensor_id++;
         }
```

Comment:
And then this. if the count is 0, set the index to 1. did someone forget php is a zero based language? It's the only way i can make it make sense. @HolgerHees thoughts?